systemdspawner/systemd.py

```diff
@@ -143,6 +144,20 @@ async def start_transient_service(
         )
         run_cmd.append(f"--property=EnvironmentFile={environment_file}")
 
+    # make sure cmd[0] is absolute, which is a requirement of systemd-run
+    # if it's relative, resolve it with Spawner's $PATH,
+    # as any other Spawner would behave
+    if not os.path.isabs(cmd[0]):
+        path = os.getenv("PATH", os.defpath)
+        path = (environment_variables or {}).get("PATH")
+        if environment_variables and "PATH" in environment_variables:
+            path = environment_variables["PATH"]
+        exe = cmd[0]
+        abs_exe = shutil.which(exe, path=path)
```

Comment:
```suggestion
        exe = cmd[0]
        path = (environment_variables or {}).get("PATH")
        abs_exe = shutil.which(exe, path=path)
```
Not entirely sure about the intention, but the result from `os.getenv` would always be discarded for the value from `environment_variables`, which would in most cases probably result in `None` from `.get`, which is fine, since `None` is also the default for `shutil.which`, making it resolve relative to `PATH`, so this should be the shortest version.

Reply:
Yes, thanks! shutil.which handles the `$PATH or os.defpath` if path is None, so this should work.